parquet-column/src/main/java/org/apache/parquet/column/impl/ColumnWriterBase.java

```diff
@@ -305,6 +305,9 @@ long getRowsWrittenSoFar() {
    * Writes the current data to a new page in the page store
    */
   void writePage() {
+    if (valueCount == 0) {
+      throw new ParquetEncodingException("writing empty page");
```

Comment:
This leaves it up to the clients to handle right? Should this be transparent to the clients / only dealt with on the parquet side?

Reply:
@vinooganesh, I think this is intended as a sanity check to ensure we don't write empty pages. The new `flush` call in `MessageColumnIO` should fix the problem.

Reply:
@rdblue , @gszadovszky ... hi, sorry for writing this way, I think that probably write here direct to code would reach authors (as I'm starting to be despair :D ) .. I hit this error and don't know how to deal with it, I'm using it in MapReduce and do nothing special, provide WritteSupport and that's all...unfortunately I'm not able to reproduce it in any test,all test with my WriteSupport class passes ... here is my [ question on StackOverFlow](https://stackoverflow.com/questions/61776917/parquet-writer-org-apache-parquet-io-parquetencodingexception-writing-empty-pa) ... thanks for any help

Reply:
> @rdblue , @gszadovszky ... hi, sorry for writing this way, I think that probably write here direct to code would reach authors (as I'm starting to be despair :D ) .. I hit this error and don't know how to deal with it, I'm using it in MapReduce and do nothing special, provide WritteSupport and that's all...unfortunately I'm not able to reproduce it in any test,all test with my WriteSupport class passes ... here is my [ question on StackOverFlow](https://stackoverflow.com/questions/61776917/parquet-writer-org-apache-parquet-io-parquetencodingexception-writing-empty-pa) ... thanks for any help

I encountered the same problem, how to solve it?

Reply:
You may check my answer at the referenced StackOverFlow thread.